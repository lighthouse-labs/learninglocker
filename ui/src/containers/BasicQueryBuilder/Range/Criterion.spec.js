import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { withInsertCSS } from 'ui/utils/hocs';
import { Map } from 'immutable';
import { initialSections } from 'ui/redux/modules/queryBuilder';
import CriterionWithStyles, { Criterion } from './Criterion.js';

const WrappedCriterion = withInsertCSS(CriterionWithStyles);

test('Criterion should render', () => {
  const section = initialSections.getIn(['result', 'children', 'scaled']);
  const criterion = ReactTestRenderer.create(
    <WrappedCriterion
      section={section} />
  ).toJSON();

  expect(criterion).toMatchSnapshot();
});

test('Criterion with value', () => {
  const section = initialSections.getIn(['result', 'children', 'scaled']);

  const mockCriteria = new Map({
    'statement.result.score.scaled': new Map({
      $gt: 0.6
    })
  });

  const criterion = ReactTestRenderer.create(
    <WrappedCriterion
      criterion={mockCriteria}
      section={section} />
  ).toJSON();

  expect(criterion).toMatchSnapshot();
});

test('Criterion operator change', () => {
  const mockMerge = jest.fn();

  const onCriterionChange = jest.fn().mockReturnValueOnce({
    merge: mockMerge
  });

  const testCriterion = new Criterion({
    criterion: new Map({
      $comment: 'A comment',
    }),
    section: initialSections.getIn(['result', 'children', 'scaled']),
    onCriterionChange,
    onDeleteCriterion: jest.fn()
  });


  testCriterion.changeOperator('<=');

  expect(onCriterionChange.mock.calls.length).toEqual(1);
  expect(onCriterionChange.mock.calls[0][0].get('statement.result.score.scaled')).toEqual(new Map({ $lte: '' }));
});

test('Criterion value change', () => {
  const mockMerge = jest.fn();

  const onCriterionChange = jest.fn().mockReturnValueOnce({
    merge: mockMerge
  });

  const testCriterion = new Criterion({
    criterion: new Map({
      $comment: 'A comment',
    }),
    section: initialSections.getIn(['result', 'children', 'scaled']),
    onCriterionChange,
    onDeleteCriterion: jest.fn()
  });


  testCriterion.handleValueChange({
    target: {
      value: '0.5'
    }
  });

  expect(onCriterionChange.mock.calls.length).toEqual(1);
  expect(onCriterionChange.mock.calls[0][0].get('statement.result.score.scaled')).toEqual(new Map({ $gt: 0.5 }));
});
