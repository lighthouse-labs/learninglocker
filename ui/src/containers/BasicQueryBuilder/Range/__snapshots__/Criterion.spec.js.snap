// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Criterion should render 1`] = `
<div
  className="criterion"
>
  <div
    className="criterionOperator"
  >
    <div
      className="select"
      onBlur={[Function]}
      onFocus={[Function]}
      tabIndex={0}
    >
      <div
        className="content btn btn-default btn-xs"
      >
        <div
          className="value"
        >
          &gt;
        </div>
        <div
          className="indicator"
        >
          <div
            className="ion-arrow-up-b"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="criterionValue noCriteria"
  >
    <input
      max={1}
      min="0"
      onChange={[Function]}
      step={0.1}
      type="number"
      value=""
    />
  </div>
</div>
`;

exports[`Criterion with value 1`] = `
<div
  className="criterion"
>
  <div
    className="criterionOperator"
  >
    <div
      className="select"
      onBlur={[Function]}
      onFocus={[Function]}
      tabIndex={0}
    >
      <div
        className="content btn btn-default btn-xs"
      >
        <div
          className="value"
        >
          &gt;
        </div>
        <div
          className="indicator"
        >
          <div
            className="ion-arrow-up-b"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="criterionValue noCriteria"
  >
    <input
      max={1}
      min="0"
      onChange={[Function]}
      step={0.1}
      type="number"
      value=""
    />
  </div>
</div>
`;
